generate_figures/utilities.py

```diff
@@ -0,0 +1,55 @@
+import pandas as pd
+import wandb
+
+def plot_params(figsize, fontsize, ticksize):
+    params = {
+        "text.usetex": True,
+        "font.family": "serif",
+        "figure.figsize": figsize,
+        "figure.dpi": 80,
+        "figure.edgecolor": "k",
+        "font.size": fontsize,
+        "axes.labelsize": fontsize,
+        "axes.titlesize": fontsize,
+        "xtick.labelsize": ticksize,
+        "ytick.labelsize": ticksize,
+    }
+    return(params)
+
+
+
+def get_results_by_tag(project, tag, test_results = False):
+    api = wandb.Api()
+    count=0
+    runs = api.runs(project, filters={"tags": tag})
+    df_test_results = pd.DataFrame()
+    for i in range(len(runs)):
+        count+=1
+        df = pd.DataFrame()
+        cols = dict(runs[i].summary)
+        cols.update(dict(runs[i].config))
+        cols['id'] = runs[i].id
+        df = pd.DataFrame.from_dict(cols, orient='index').T
+ 
+        if test_results:
+            reference_path = f"{project}/run-{cols['id']}-test_results:v0"
+            test_results = api.artifact(str(reference_path)).get("test_results")
+            data = pd.DataFrame(data = test_results.data, columns = test_results.columns)
+            df = pd.concat([df, data], axis = 1)
```

Comment:
I assume that this is tested carefully because it is different than what we had before?  Before we didn't concatenate it like this but instead we held off on creating the dataframe until the end and added in columns as required if the test_results were being used?